src/ILLink.Shared/TrimAnalysis/NullableRuntimeSystemTypeHandleValue.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using ILLink.Shared.DataFlow;
+using ILLink.Shared.TypeSystemProxy;
+
+namespace ILLink.Shared.TrimAnalysis
+{
+	/// <summary>
+	/// This represents a type handle Nullable<T> where T is a known SystemTypeValue.
+	/// It is necessary to track the underlying type to propagate DynamicallyAccessedMembers annotations to the underlying type when applied to a Nullable.
+	/// </summary>
+	sealed record NullableRuntimeSystemTypeHandleValue : SingleValue
```

Comment:
Same question about `SystemTypeValue` and `RuntimeTypeHandleValue` - Linker currently tries to conform to static types - meaning that `SystemTypeValue` has static type of `Type`, while `RuntimeTypeHandleValue` is `RuntimeTypeHandle`. Otherwise they're identical.
I guess we could merge them.... this was carried over from .NET Native - and it's true that linker still holds onto the static typic pretty much always.
I'm fine keeping this as is - if we don't like it, let's make it a different change which applies to non-Nullable as well.

Reply:
This class is to `NullableSystemTypeValue` as `RuntimeTypeHandleValue` is to the `SystemTypeValue` (Nullable<T> vs typeof(Nullable<T>)). My understanding is that theoretically yes these could be merged, but it might be helpful to keep them separate to more accurately model the program being analyzed.